tf2/include/tf2/time.h

```diff
@@ -58,11 +61,15 @@ namespace tf2
 
   inline std::string displayTimePoint(const TimePoint& stamp)
   {
-    std::time_t time = std::chrono::system_clock::to_time_t(std::chrono::time_point_cast<std::chrono::milliseconds>(stamp));
-    char str[100];
-    std::strftime(str, sizeof(str), "%c", std::localtime(&time));
-    return std::string(str);
+    const char * format_str = "%.6f";
+    double current_time = timeToSec(stamp);
+    int buff_size = snprintf(NULL, 0, format_str, current_time);
+    char * buffer = new char[buff_size];
```

Comment:
new throws bad_alloc when it fails: http://www.cplusplus.com/reference/new/bad_alloc/